evm/src/cpu/kernel/asm/halt.asm

```diff
@@ -1,6 +1 @@
 global halt:
```

Comment:
We must put a PANIC instruction below `halt` to ensure that the prover cannot choose to _not_ halt at this address (in other words, it could execute whatever comes after `halt` in memory).

```suggestion
global halt:
    PANIC
```

Reply:
Oh right! good catch